scala.wake

```diff
@@ -44,11 +44,9 @@ def coursierBin =
     Nil      = fetchCoursier Unit
     l        = makeBadPath (makeError "Multiple coursier executables found! {catWith ", " l}")
 
-# Only get ivydependencies.json files 0 or 1 levels under workspace root
-# This emulates Wit <= 0.12 behavior where packages are placed directly below
-# the workspace root and Wit fetches dependencies in ivydependencies.json
-# located at the root of packages.
-def ivyDepsFiles = sources "." `[^/]*/?ivydependencies\.json`
+def ivyDepsFiles =
+  sources "." `.*/?ivydependencies\.json`
+  | filter (!matches `.*/tests/.*` _.getPathName)
```

Comment:
This approach is purely intended as a temporary measure for Wit version migration. If we're looking at supporting an alternative repository layout, I'd rather just rethink the approach. I really don't like grabbing all of the files except those in "tests" directories. I'd rather use `.wakeignore` for excluding directories if we're going to do any sort of globbing.

Possibilities:
1. Ivy dependencies and coursier are resources that should be provided by environment package (with utilities to help download dependencies)
2. Ivy dependencies are automatically downloaded when a `ScalaModule` is created
    * If the user wants prefetching, they should declare their dependencies in `preinstall`

Reply:
I've tried adding a publish/subscribe to this, without removing the old stuff

In rocket-chip for example, a wake file could contain
`publish ivyDepLocations = rocketChipRoot, Nil`

How do you feel about this approach?